xds/src/test/java/io/grpc/xds/ClientXdsClientV3Test.java

```diff
@@ -286,6 +298,15 @@ protected Message buildListenerForRds(String name, String rdsResourceName) {
           .build();
     }
 
+    @Override
+    protected Message buildListenerInvalid(String name) {
```

Comment:
The V2 and V3 versions look identical. Why can't this be part of Base?

Reply:
v2 and v3 uses different `Listener`. I guess you are saying passing a boolean argument like `forV3` and the method would be something like

```
if (forV3) {
  // build v3 Listener
} else {
  // build v2 Listener
}
```

You don't write significantly less amount of code by doing so, there is barely no code reuse. Also, this is generally [bad/anti-pattern](https://blog.iannelson.uk/back-to-basics-on-the-use-and-abuse-of-the-humble-boolean/), although it's just a helper method in the test.

Reply:
Oops, this is as per the existing pattern - part of the `MessageFactory`. So that's okay. I thought otherwise.